### PR TITLE
keyd: 2.4.2 -> 2.4.3

### PR DESCRIPTION
**Pull request by JohnAZoidberg**
###### Description of changes

https://github.com/rvaiya/keyd/releases/tag/v2.4.3

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by JohnAZoidberg** [comment]
Seems to break the `keyd` service. I'm investigating.

**Comment by peterhoeg** [comment]
off-topic, but I’m using `evscript' which works great for my simple needs.

**Comment by JohnAZoidberg** [comment]
Works now. The program calls `nice` which was blocked by systemd.
cc module author @woojiq

**Comment by woojiq** [comment]
@JohnAZoidberg haven't tested it manually, but if it passes the qemu test it's probably fine. Have you read this comment:
https://github.com/NixOS/nixpkgs/blob/3e01b0434e7a1777b421c1e2571de6cbeaa06656/nixos/modules/services/hardware/keyd.nix#L115-L119
See: https://github.com/NixOS/nixpkgs/pull/221321#issuecomment-1479471297
It's not mine, so does the new release affect it somehow?

**Comment by JohnAZoidberg** [comment]
> It's not mine, so does the new release affect it somehow?

Hmmm, if we change it at compile time it won't run from commandline anymore. Only in the service.
Since the keyd folder doesn't exit. I guess it's okay actually, since you'd not usually run it manually.

File: nixos/modules/services/hardware/keyd.nix
Comment (woojiq) [inline nixos/modules/services/hardware/keyd.nix:1]: I've tested manually too and it seems useless now. Though you are probably more experienced with nix so make final decision, xd.
```diff
- # this is configurable in 2.4.2, later versions seem to remove this option.
- # post-2.4.2 may need to set makeFlags in the derivation:
- #
- #     makeFlags = [ "SOCKET_PATH/run/keyd/keyd.socket" ];
- environment.KEYD_SOCKET = "/run/keyd/keyd.sock";
-
```

**Review by maxammann — APPROVED** [review]
Looks good to me. Can we merge this?

`keyd` on 2.4.2 has a bug where it takes up 100% of the CPU. I believe this is fixed in the newer version.

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 245327` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>keyd</li>
  </ul>
</details>

nixosTests.keyd also passes. LGTM

**Pull request merged**
